tools/juliapkg/src/result.jl

```diff
@@ -649,10 +649,11 @@ end
 # execute background tasks in a loop, until task execution is finished
 function execute_tasks(state::duckdb_task_state, con::Connection)
     while !duckdb_task_state_is_finished(state)
-        GC.safepoint()
         duckdb_execute_n_tasks_state(state, 1)
         if duckdb_execution_is_finished(con.handle)
             break
+            Base.yield()
```

Comment:
This is much more expensive than `GC.safepoint`, perhaps it can be called every 100k iterations or so, depending on the perf impact?

Reply:
`break` comes before `Base.yield` and `GC.safepoint()`?

Reply:
Good point, I guess yield and safepoint should've been outside of the if statement?

Reply:
Executing a task in DuckDB can take quite some time and process hundreds of thousands/millions of tuples (depending on data set size). Perhaps that is already the correct granularity for `Base.yield`? We could run some benchmarks.

Definitely should be outside of the if statement.